### PR TITLE
Add documentation on legacy kubernetes_auth file to kubeconfig-file.md

### LINKED ISSUE #3869: kubernetes_auth missing documentation
After installing nginx on the kubernetes master (currently non-k8s docker container) and forwarding external http requests  to the kubernetes-ro service through nginx with simple htpasswd auth, I tested it (to use within CI tool), but it failed to pass nginx.

```
/tmp ❯❯❯ master_ip=54.X.X.X
/tmp ❯❯❯ wget https://storage.googleapis.com/kubernetes-release/release/v0.8.1/bin/linux/amd64/kubectl
--2015-01-28 23:53:43--  https://storage.googleapis.com/kubernetes-release/release/v0.8.1/bin/linux/amd64/kubectl
Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.235.106, 74.125.235.108, 74.125.235.107
Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.235.106|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 12221288 (12M) [application/octet-stream]
Saving to: ‘kubectl’

kubectl                                         100%[========================================================================================================>]  11.65M  5.18MB/s   in 2.2s

2015-01-28 23:53:45 (5.18 MB/s) - ‘kubectl’ saved [12221288/12221288]

/tmp ❯❯❯ chmod +x kubectl                                                                                                                                                                    
/tmp ❯❯❯ ./kubectl get --insecure-skip-tls-verify=true -s $master_ip -a ~/.kubernetes_auth pods
F0128 23:54:34.151603   17748 get.go:75] request [&{Method:GET URL:http://54.X.X.X/api/v1beta1/pods?namespace=default Proto:HTTP/1.1 ProtoMajor:1 ProtoMinor:1 Header:map[] Body:<nil> ContentLength:0 TransferEncoding:[] Close:false Host:54.X.X.X Form:map[] PostForm:map[] MultipartForm:<nil> Trailer:map[] RemoteAddr: RequestURI: TLS:<nil>}] failed (401) 401 Unauthorized: <html>
<head><title>401 Authorization Required</title></head>
<body bgcolor="white">
<center><h1>401 Authorization Required</h1></center>
<hr><center>nginx/1.6.2</center>
</body>
</html>

/tmp ❯❯❯ curl -u $(cat ~/.kubernetes_auth) $master_ip
<html><body>Welcome to Kubernetes</body></html>%                                                                                                                                               /tmp ❯❯❯
```

Is it because we are required to use https when querying the apiserver using kubectl? Or because the -a option is used for a different auth ?
How would you solve this issue ?

Thank you in advance!

### PR DESCRIPTION
**Pull request by j3ffml**
Closes #3869. cc @deads2k

### CODE REVIEW

File: docs/kubeconfig-file.md [hunk 1]
```diff
@@ -46,14 +46,46 @@ users:
...
+  "User": "admin",
+  "Password": "secret",
+  "CertFile": "/path/to/my/client/cert",
+  "KeyFile": "/path/to/my/client/key",
+  "CAFile": "/path/to/my/server/cafile",
+  "BearerToken": "secrettoken",
+  "Insecure": false
+}
+```
+All entries are optional. `User`, `Password`, `CertFile`, `KeyFile`, and `BearerToken` are applied to the kubectl user. `CAFile` and `Insecure` apply to the cluster. The same rules regarding one authentication technique apply as for .kubeconfig entries (see below).
```
Comment (deads2k) [inline docs/kubeconfig-file.md]: validateAuthInfo doesn't detect conflicting `BearerToken` and `User` authentication loaded from `--auth-path`: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/clientcmd/validation.go#L139

File: docs/kubeconfig-file.md [hunk 2]
```diff
@@ -46,14 +46,46 @@ users:
...
+```
+users:
+- name: black-user
+  user:
+    auth-path: path/to/my/existing/.kubernetes_auth_file
+```
+
+The entries in a file loaded via auth-path will be applied to both the user and cluster of the current context.
+
+**WARNING**: merging auth from a mixture of kubernetes_auth file entries and .kubeconfig user entries is hard to debug and should be avoided. kubernetes_auth file support exists mostly for tests and is being deprecated.
```
Comment (deads2k) [inline docs/kubeconfig-file.md]: Lead the section with this?

**Comment by erictune** [comment]
I'll merge this once @deads2k gives the LGTM

File: docs/kubeconfig-file.md (hunk 1 above)
Comment (jlowdermilk) [inline docs/kubeconfig-file.md]: Changed wording to warn against setting both.

File: docs/kubeconfig-file.md (hunk 2 above)
Comment (jlowdermilk) [inline docs/kubeconfig-file.md]: done

**Comment by jlowdermilk** [comment]
PTAL

**Comment by deads2k** [comment]
lgtm

**Pull request merged**